npm_and_yarn/lib/dependabot/npm_and_yarn/file_parser.rb

```diff
@@ -98,23 +109,36 @@ def manifest_dependencies
         dependency_set
       end
 
+      sig { returns(LockfileParser) }
       def lockfile_parser
-        @lockfile_parser ||= LockfileParser.new(
-          dependency_files: dependency_files
-        )
+        @lockfile_parser ||= T.let(LockfileParser.new(
+                                     dependency_files: dependency_files
+                                   ), T.nilable(Dependabot::NpmAndYarn::FileParser::LockfileParser))
       end
 
+      sig { returns(Dependabot::FileParsers::Base::DependencySet) }
       def lockfile_dependencies
         lockfile_parser.parse_set
       end
 
+      sig do
+        params(file: DependencyFile, type: T.untyped, name: String, requirement: String)
+          .returns(T.nilable(Dependency))
+      end
       def build_dependency(file:, type:, name:, requirement:)
         lockfile_details = lockfile_parser.lockfile_details(
           dependency_name: name,
           requirement: requirement,
           manifest_name: file.name
         )
         version = version_for(requirement, lockfile_details)
+        converted_version = T.let(if version.nil?
```

Comment:
This part got a bit fiddly because we seem to have been passing some possible bare Gem::Version instances before.

Reply:
There is a class hierarchy of `Gem::Version` -> `Dependabot::Version` -> `Dependabot::NpmAndYarn::Version`